src/slack.coffee

```diff
@@ -17,33 +15,31 @@ class SlackBot extends Adapter
     exitProcessOnDisconnect = !!process.env.HUBOT_SLACK_EXIT_ON_DISCONNECT
 
     # Take our options from the environment, and set otherwise suitable defaults
-    options =
+    @options =
+      autoMark: true
+      logLevel: 'error'
+      dataStore: new MemoryDataStore()
       token: process.env.HUBOT_SLACK_TOKEN
       autoReconnect: !exitProcessOnDisconnect
-      autoMark: true
-      exitOnDisconnect: exitProcessOnDisconnect
-      proxyUrl: process.env.https_proxy
 
```

Comment:
I think the change in the options means this qualifies as a breaking change, and hence will require us to bump the major version number.